src/DaemonExec.ts

```diff
@@ -0,0 +1,138 @@
+import * as Docker from 'dockerode'
+
+interface ContainerModem {
+	demuxStream(
+		stream: NodeJS.ReadWriteStream,
+		stdoutStream: NodeJS.WritableStream,
+		stderrStream: NodeJS.WritableStream,
+	): void
+}
+
+/**
+ * Returns information about the images
+ * @param docker - The docker daemon to pull information from
+ * @returns an array of ImageInfo
+ */
+export async function listImages(docker: Docker): Promise<Docker.ImageInfo[]> {
+	const images = await docker.listImages({})
+	return images
+}
+
+/**
+ * Returns information about the containers
+ * @param docker - The docker daemon to pull information from
+ * @returns an array of ContainerInfo
+ */
+export async function listContainers(
+	docker: Docker,
+): Promise<Docker.ContainerInfo[]> {
+	const containers = await docker.listContainers()
+	return containers
+}
+
+/**
+ * Starts the import of an image
+ * @param docker - The docker daemon with which the image will be stored
+ * @param name - The name of the image to pull
+ * @returns a ReadableStream to track the progress of the import
+ */
+export async function importImage(
+	docker: Docker,
+	name: string,
+): Promise<NodeJS.ReadWriteStream> {
+	const stream: NodeJS.ReadWriteStream = await (docker.pull(
+		name,
+	) as Promise<NodeJS.ReadWriteStream>)
+	return stream
+}
+
+/**
+ * Creates a Docker container
+ * @param docker - the docker daemon that controls the container
+ * @param image - the image the container uses
+ * @param command - an array that holds the command for the container
+ * @param volumePairs - an array of string pairs that represents the volumes
+ * @returns the Container
+ */
+export async function createContainer(
+	docker: Docker,
+	image: string,
+	command: string[],
+	volumePairs: [string, string][] = [],
```

Comment:
Looking at this function, the caller doesn't know what to provide here (I'm assuming it's 2-arrays of `[from path, to path]`). Can you export an interface like:

```ts
interface VolumeDefinition {
	fromPath: string
	toPath: string
}
```

And use an array of that as the parameter? This would also require changing the use of this function to use an object with named keys rather than a 2-array.